robottelo/hosts.py

```diff
@@ -0,0 +1,368 @@
+import logging
+from urllib.parse import urljoin
+from urllib.parse import urlunsplit
+
+from broker.hosts import Host
+
+from robottelo.config import settings
+from robottelo.constants import DISTRO_RHEL6
+from robottelo.constants import DISTRO_RHEL7
+from robottelo.constants import REPOS
+from robottelo.helpers import install_katello_ca
+from robottelo.helpers import remove_katello_ca
+
+logger = logging.getLogger(__name__)
+
+
+class ContentHostError(Exception):
+    pass
+
+
+class ContentHost(Host):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.run = self.execute
+
+    def download_install_rpm(self, repo_url, package_name):
+        """Downloads and installs custom rpm on the virtual machine.
+
+        :param repo_url: URL to repository, where package is located.
+        :param package_name: Desired package name.
+        :return: None.
+        :raises robottelo.hosts.ContentHostError: If package wasn't installed.
+
+        """
+        self.execute(f'wget -nd -r -l1 --no-parent -A \'{package_name}.rpm\' {repo_url}')
```

Comment:
Do you have to use wget here? It, wget, is not installed by default IIRC. Using curl -O is fine if its just one file to be downloaded. But curl does not support recursive download.

Reply:
if this is something curl can't do, then we'll need to have @bherrin3 add it to our template

Reply:
Alternately, just have a "list of changes" change doc or a readme file listing all changes made in the creation of a template. Then snap templates can have time saving additions while GA templates can be "as released". Given enough transparency, the user can assess how the changes impact their test scenario.